build-android/jni/Android.mk

```diff
@@ -0,0 +1,101 @@
+# Copyright 2015 The Android Open Source Project
+# Copyright (C) 2015 Valve Corporation
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#      http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+LOCAL_PATH := $(call my-dir)
+SRC_DIR := ../..
+LAYER_DIR := ../generated
+THIRD_PARTY := ../third_party
+
+VULKAN_INCLUDE := $(LOCAL_PATH)/$(THIRD_PARTY)/Vulkan-Headers/include
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := extlayer_utils
+
+LOCAL_SRC_FILES += $(SRC_DIR)/utils/allocator.cpp
+LOCAL_SRC_FILES += $(SRC_DIR)/utils/vk_format_utils.cpp
+LOCAL_SRC_FILES += $(SRC_DIR)/utils/vk_layer_config.cpp
+LOCAL_SRC_FILES += $(SRC_DIR)/utils/generated/vk_safe_struct.cpp
+LOCAL_SRC_FILES += $(SRC_DIR)/utils/generated/lvt_function_pointers.cpp
+
+LOCAL_C_INCLUDES += $(VULKAN_INCLUDE) \
+                    $(LOCAL_PATH)/$(SRC_DIR)/utils/generated \
+                    $(LOCAL_PATH)/$(SRC_DIR)/utils
+LOCAL_CPPFLAGS += -std=c++11 -Wall -Werror -Wno-unused-function -Wno-unused-const-variable -fexceptions
+LOCAL_CPPFLAGS += -DVK_ENABLE_BETA_EXTENSIONS -DVK_USE_PLATFORM_ANDROID_KHR -DVK_PROTOTYPES -fvisibility=hidden
+include $(BUILD_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := VkLayer_khronos_synchronization
+LOCAL_SRC_FILES += $(SRC_DIR)/layers/synchronization2.cpp
+LOCAL_C_INCLUDES += $(VULKAN_INCLUDE) \
+                    $(LOCAL_PATH)/$(SRC_DIR)/utils \
+                    $(LOCAL_PATH)/$(SRC_DIR)/utils/generated \
+                    $(LOCAL_PATH)/$(THIRD_PARTY)/shaderc/third_party/spirv-tools/external/spirv-headers/include
+LOCAL_STATIC_LIBRARIES += extlayer_utils glslang SPIRV-Tools SPIRV-Tools-opt
+LOCAL_CPPFLAGS += -std=c++11 -Wall -Werror -Wno-unused-function -Wno-unused-const-variable -fexceptions
```

Comment:
I needed `-Wno-cast-calling-convention` or else the 32-bit build got mad about `ADD_HOOK`